src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.ConnectionManager.cs

```diff
@@ -87,25 +87,39 @@ public Task<Connection> TryCreateConnectionAsync(string serviceName, object call
                 return TryGetConnectionFromPoolAsync(serviceName, cancellationToken);
             }
 
-            private async Task<Connection> TryGetConnectionFromPoolAsync(string serviceName, CancellationToken cancellationToken)
+            private async Task<OwnedDisposable<Connection>> TryGetConnectionFromPoolAsync(string serviceName, CancellationToken cancellationToken)
             {
-                var queue = _pools.GetOrAdd(serviceName, _ => new ConcurrentQueue<JsonRpcConnection>());
+                var queue = _pools.GetOrAdd(serviceName, _ => new ConcurrentQueue<OwnedDisposable<Connection>>());
                 if (queue.TryDequeue(out var connection))
                 {
-                    return new PooledConnection(this, serviceName, connection);
+                    try
+                    {
+                        return new OwnedDisposable<Connection>(() => new PooledConnection(this, serviceName, ref connection));
```

Comment:
💭 @mavasani The factory function avoids the need to transfer an object into this constructor, but it comes with an allocation cost.